sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/authentication/ContainerRegistryTokenService.java

```diff
@@ -80,7 +84,12 @@ public Mono<AccessToken> getToken(TokenRequestContext tokenRequestContext) {
             }
 
             return this.refreshTokenCache.getToken(requestContext)
-                .flatMap(refreshToken -> this.tokenService.getAcrAccessTokenAsync(refreshToken.getToken(), scope, serviceName, TokenGrantType.REFRESH_TOKEN));
+                .flatMap(refreshToken -> this.tokenService.getAcrAccessTokenAsync(refreshToken.getToken(), scope,
+                    serviceName, TokenGrantType.REFRESH_TOKEN));
         }).doOnError(err -> logger.error("Could not fetch the ACR error token.", err));
     }
+
+    public AccessToken getTokenSync(TokenRequestContext tokenRequestContext) {
+        return this.getToken(tokenRequestContext).block();
```

Comment:
we should have todo/work item to revisit this later. this one seems to be simpler than identity.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/29746